### DISCUSSION TITLE
I want to make a package that namespaces `avro`

### DISCUSSION THREAD

**Discussion by baseplate-admin** [pyinstaller/pyinstaller#8718] [Help] [answered]
Hi, we have a package, [`avro.py`](https://github.com/hitblast/avro.py) that uses the same namespace as [`avro`](https://pypi.org/project/avro/). When invoking pyinstaller, it looks for [`hooks-avro.py`](https://github.com/pyinstaller/pyinstaller-hooks-contrib/blob/7e6c20f983aae10f3750766adc2efc89bc100c42/_pyinstaller_hooks_contrib/stdhooks/hook-avro.py).

So, my question is:
1. Can we modify the hook to pick up `avro.py` instead of `avro`?
2. Can we exclude the hook's execution?

cc : @hitblast

**Comment by bwoodsend** [comment]
This question doesn't make a lot of sense but I'm guessing you mean that you have a package with the same importable name as `avro` and you want to prevent the `hook-avro.py` (which is intended for the other `avro`) from running?

**Reply by baseplate-admin** [reply]
> This question doesn't make a lot of sense but I'm guessing you mean that you have a package with the same importable name as avro and you want to prevent the hook-avro.py (which is intended for the other avro) from running? 

More or less yes.

**Reply by bwoodsend** [reply]
I guess it's high time that we started putting guards in hooks that check that the distribution that they're targeting is installed...

**Reply by rokm** [reply]
True, although that won't completely solve the problem. It might help avoiding errors when the expected distribution is unavailable, but if you have an eponymous private package that also requires some hook functionality, you will still need to override the hook. And I suppose name clashes are less common with "public" packages (e.g., PyPI-published), because wider exposure ensures in issues being reported against one of the distributions.

**Comment by rokm** [comment, marked as answer]
You can override any standard hook from PyInstaller or pyinstaller-hooks-contrib by creating your own hook, and passing its parent directory to PyInstaller via [`--additional-hooks-dir`](https://pyinstaller.org/en/stable/usage.html#cmdoption-additional-hooks-dir).

E.g., create a directory called `my-hooks`, and create `hook-avro.py` in it. Then pass the path to your custom hooks directory to PyInstaller via `--additional-hooks-dir my-hooks`. Your custom hook will then be ran instead of the one provided by pyinstaller-hooks-contrib. You can make it completely no-op (leave it empty), or customize it to your package, if necessary.

**Reply by baseplate-admin** [reply]
> E.g., create a directory called my-hooks, and create hook-avro.py in it. Then pass the path to your custom hooks directory to PyInstaller via --additional-hooks-dir my-hooks. Your custom hook will then be ran instead of the one provided by pyinstaller-hooks-contrib. You can make it completely no-op (leave it empty), or customize it to your package, if necessary.

Worked

Thanks a bunch
